spec/inputs/tcp_spec.rb

```diff
@@ -52,260 +52,293 @@ def get_port
     end
   end
 
-  it "should read plain with unicode" do
-    event_count = 10
-    conf = <<-CONFIG
-      input {
-        tcp {
-          port => #{port}
-        }
-      }
-    CONFIG
-
-    host = 'localhost'
-    events = input(conf) do |pipeline, queue|
-      socket = Stud::try(5.times) { TCPSocket.new(host, port) }
-      event_count.times do |i|
-        # unicode smiley for testing unicode support!
-        socket.puts("#{i} ☹")
-        socket.flush
+  [:disabled,:v1].each do |ecs_compatibility|
+    context "with `ecs_compatibility => `#{ecs_compatibility}`" do
+      before(:each) do
+        allow_any_instance_of(described_class).to receive(:ecs_compatibility).and_return(ecs_compatibility)
       end
-      socket.close
 
-      event_count.times.collect {queue.pop}
-    end
+      it "should read plain with unicode" do
+        event_count = 10
+        conf = <<-CONFIG
+          input {
+            tcp {
+              port => #{port}
+            }
+          }
+        CONFIG
+
+        host = 'localhost'
+        events = input(conf) do |pipeline, queue|
+          socket = Stud::try(5.times) { TCPSocket.new(host, port) }
+          event_count.times do |i|
+            # unicode smiley for testing unicode support!
+            socket.puts("#{i} ☹")
+            socket.flush
+          end
+          socket.close
 
-    insist { events.length } == event_count
-    events = events.sort_by {|e| e.get("message")} # the ordering of events in the queue is highly timing-dependent
-    event_count.times do |i|
-      event = events[i]
-      insist { event.get("message") } == "#{i} ☹"
-      insist { ["localhost","ip6-localhost"].includes? event.get("host") }
-      insist { event.get("[@metadata][ip_address]") } == '127.0.0.1'
-    end
-  end
+          event_count.times.collect {queue.pop}
+        end
 
-  it "should handle PROXY protocol v1 connections" do
-    event_count = 10
-    conf = <<-CONFIG
-      input {
-        tcp {
-          proxy_protocol => true
-          port => '#{port}'
-        }
-      }
-    CONFIG
-
-    events = input(conf) do |pipeline, queue|
-      socket = Stud::try(5.times) { TCPSocket.new("127.0.0.1", port) }
-      socket.puts("PROXY TCP4 1.2.3.4 5.6.7.8 1234 5678\r");
-      socket.flush
-      event_count.times do |i|
-        # unicode smiley for testing unicode support!
-        socket.puts("#{i} ☹")
-        socket.flush
+        insist { events.length } == event_count
+        events = events.sort_by {|e| e.get("message")} # the ordering of events in the queue is highly timing-dependent
+        event_count.times do |i|
+          event = events[i]
+          insist { event.get("message") } == "#{i} ☹"
+
+          case ecs_compatibility
+          when :disabled
```

Comment:
`ecs_compatibility` is used only 2 tests inside the all test cases, and those places are where it's used 
```ruby
case ecs_compatibility
when :disabled ...
when :v1 ...
end
```
I was wondering if:
- the use of `ecs-compatibility` setting has meaning also for the remaining tests?
- the common part of the tests could be extracted in a shared example or some test helper that could share the common part and accept a block to verify the expectations in the 2 cases?
The motivation is that use `if`s and `loop`s in test code make it more unreadable

Reply:
I've restructured to (a) use the rspec helpers we published in v1.2 of the support gem, and (2) rearranged so that only those that are liable to have differing behaviour are ent through the variants.